### PR TITLE
Ignore Health and Metrics artifacts

### LINKED ISSUE #22437: Ignore Health and Metrics artifacts
### Description

When Health(`health-enabled`) and Metrics (`metrics-enabled`)  are disabled, we could ignore their artifacts.

### PR DESCRIPTION
**Pull request by mabartos**
Closes #22437

### CODE REVIEW

**Comment by mabartos** [comment]
@vmuzikar @ahus1 Could you please check it? Thanks

**Review by ahus1 — COMMENTED** [review]
I saw the reasons when we did this for database drivers which we excluded from the distribution and when we wanted to avoid the warnings in the log.

I am not sure about the reasons here and what benefits we're expecting from this change. Is is a shorter startup time, and less memory use? This comes with the trade-off of maintainability, as we would need to know about the transitional dependencies of those libraries. 

If the benefits are not significant, I'd recommend to keep the code simpler to maintain by not implementing this change.

If the CND team still wants to merge it, I won't veto it, so that's why I just comment here.

**Comment by vmuzikar** [comment]
@mabartos Could you please rebase?

**Comment by mabartos** [comment]
@vmuzikar I marked the PR as a draft as I've encountered an issue with the micrometer. Even when the feature is disabled, there's some requirement on the Quarkus side to have particular micrometer components available in the core, which is strange. I'll analyze it once it fits my TODO plan with regard to priorities.

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/Build.java [hunk 1]
```diff
@@ -94,13 +96,10 @@ public void run() {
...
-                MicroProfileConfigProvider.NS_KEYCLOAK_PREFIX + ClassLoaderOptions.IGNORE_ARTIFACTS.getKey());
-
-        if (ignoredArtifacts != null && StringUtil.isNotBlank(ignoredArtifacts.getValue())) {
-            // ignored artifacts must be set prior to starting re-augmentation
-            System.setProperty("quarkus.class-loading.removed-artifacts", ignoredArtifacts.getValue());
-        }
+        // ignored artifacts must be set prior to starting re-augmentation
+        Optional.ofNullable(Configuration.getCurrentBuiltTimeProperty(QUARKUS_REMOVED_ARTIFACTS_PROPERTY))
+                .map(ConfigValue::getValue)
+                .ifPresent(s -> System.setProperty(QUARKUS_REMOVED_ARTIFACTS_PROPERTY, s));
```
Comment (mabartos) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/Build.java:102]: Explanation in the: https://github.com/keycloak/keycloak/pull/25012#discussion_r1405951198

**Comment by vmuzikar** [comment]
@mabartos Could you please rebase?

**Review by vmuzikar — APPROVED** [review]
LGTM, thanks @mabartos.

**Pull request merged**